client/src/components/item/itemPopUp.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { Dialog } from "primereact/dialog";
+import { Button } from "primereact/button";
+import { InputNumber } from "primereact/inputnumber";
+import IItem from "../../types/IItem";
+import { cartStore } from "../../store/cartStore";
+import { observer } from "mobx-react";
+import { toast } from 'react-toastify';
+
+import "primeicons/primeicons.css";
+import "primereact/resources/themes/saga-blue/theme.css";
+import "primereact/resources/primereact.css";
+import "primeflex/primeflex.css";
+import "../../comCss/item.css";
+
+interface ItemPopUpPtops {
+  itemPopUp: IItem | undefined;
+  setItemPopUp: React.Dispatch<React.SetStateAction<IItem | undefined>>;
+  displayItemPopUp: boolean;
+  setDisplayItemPopUp: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const ItemPopUp: React.FC<ItemPopUpPtops> = observer(({ itemPopUp,setItemPopUp,displayItemPopUp,setDisplayItemPopUp }) => {
+
+  const [unitsToAdd, setUnitsToAdd] = useState<number>(0);
+
+  const handleItemDialog = () => {
+    setDisplayItemPopUp((itemD) => !itemD);
+  };
+
+  const numUnits = (e: any) => {  
+    setUnitsToAdd(e);  
+  };
+
+  const addItemToCart = async() =>{   
+    await cartStore.addItemToCart(itemPopUp!, unitsToAdd);
```

Comment:
why the `!` ?

Reply:
if I dont write the "!" tab it creats the error: "Argument of type 'string | undefined' is not assignable to parameter of type 'string'.Type 'undefined' is not assignable to type 'string'." , so I can solve it in tow ways , put the "!" tab, or write: "itemPopUp as IItem" before I send the prop to another components.